bin/tape

```diff
@@ -41,18 +33,43 @@ if (typeof opts.ignore === 'string') {
     var matcher = ignore.createMatcher(ignoreStr);
 }
 
-opts._.forEach(function (arg) {
-    // If glob does not match, `files` will be an empty array.
-    // Note: `glob.sync` may throw an error and crash the node process.
-    var files = glob.sync(arg);
+hasDynamicImport().then(function (hasSupport) {
+    var importOrRequire = hasSupport ? require('./import-or-require') : function (file) {
+        require(file);
+    };
 
-    if (!Array.isArray(files)) {
-        throw new TypeError('unknown error: glob.sync did not return an array or throw. Please report this.');
-    }
+    var loadingPromise = opts.require
+        .filter(function (module) {
+            /* This check ensures we ignore `-r ""`, trailing `-r`, or
+             * other silly things the user might (inadvertently) be doing.
+             */
+            return module;
+        }).map(function (module) {
+            return resolveModule(module, { basedir: cwd, extensions: Object.keys(require.extensions) });
+        }).reduce(function (loadingPromise, module) {
+            return loadingPromise ? loadingPromise.then(function () {
+                return importOrRequire(module);
+            }) : importOrRequire(module);
+        }, undefined);
```

Comment:
```suggestion
        }).reduce(function (loadingPromise, module) {
            return loadingPromise.then(function () {
                return importOrRequire(module);
            });
        }, Promise.resolve());
```

Reply:
wont we loose support for Node 0.10 and earlier with this logic (they lack promise support)?